lib/services/auth.dart

```diff
@@ -26,32 +26,18 @@ Future<String> signInWithGoogle() async {
   final User user = authResult.user;
 
   if (user != null) {
-    // Checking if email and name is null
-    assert(user.email != null);
-    assert(user.displayName != null);
-    assert(user.photoURL != null);
-
-    name = user.displayName;
-    email = user.email;
-    imageUrl = user.photoURL;
-
-    // Only taking the first part of the name, i.e., First Name
-    if (name.contains(" ")) {
-      name = name.substring(0, name.indexOf(" "));
+    final QuerySnapshot result = await FirebaseFirestore.instance
+        .collection('users')
+        .where('UserId', isEqualTo: user.uid)
+        .get();
+    final List<QueryDocumentSnapshot> documents = result.docs;
+    if (documents.length == 0) {
+      FirebaseFirestore.instance.collection('users').doc(user.uid).set(
+          UserProfile.newuser(user.uid, faker.internet.userName(),
```

Comment:
there is some problem here

Reply:
new user is not defined

Reply:
I have added it now sorry for the confusion